File: files/2025-06-15/CSEC 15062025.R
library(tidyverse)
library(readxl)

path = "files/2025-06-15/Challenge 35.xlsx"
input = read_excel(path, range = "B3:B11")
test = read_excel(path, range = "D3:E4")

result = input %>%
  mutate(
    title = case_when(
      str_count(Names, "[AEIOUaeiou]") == 0 ~ "No Vowels",
      str_count(Names, "[AEIOUaieou]") == 5 ~ "All Vowels",
      TRUE ~ "Some Vowels"
    )
  ) %>%
  filter(title != "Some Vowels") %>%
  summarise(
    count = n(),
    .by = title
  ) %>%
  pivot_wider(
    names_from = title,
    values_from = count,
    values_fill = 0
  )
all.equal(result, test, check.attributes = FALSE)
# TRUE

File: files/2025-06-15/CSEC 15062025.py
import pandas as pd
import re

# Read data from Excel
path = "files/2025-06-15/Challenge 35.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=2, nrows=9)
test = pd.read_excel(path, usecols="D:E", skiprows=2, nrows=1)

input['vowel_count'] = input.iloc[:,0].astype(str).apply(lambda name: sum(1 for c in name if c.lower() in 'aeiou'))
input['title'] = input['vowel_count'].apply(lambda vowels: "No Vowel" if vowels == 0 else ("All Vowels" if vowels == 5 else "Some Vowels"))
filtered = input[input['title'] != "Some Vowels"]

pivoted = filtered.value_counts('title').reindex(["No Vowel", "All Vowels"], fill_value=0).to_frame().T
pivoted.columns.name = None
pivoted.reset_index(drop=True, inplace=True)

print(pivoted.equals(test)) # True